### ISSUE TITLE
Panic on invalid input to Image::io::Reader::decode

### ISSUE THREAD

**Issue by MinghuaWang** [image-rs/image#2128]
## Describe the bug
Panic could be triggered when an invalid input was passed to Image::io::Reader::decode().

## Expected
Not panic. It could be an error reported to the users.

## Actual behaviour
An arithmetic overflow was reported, as shown bellow. 
`thread 'main' panicked at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/lossless.rs:438:13:
attempt to add with overflow`

Fulll stack backtrace:
   0: rust_begin_unwind
             at /rustc/07688726805d5db0a4bca445a6651d09708041ea/library/std/src/panicking.rs:617:5
   1: core::panicking::panic_fmt
             at /rustc/07688726805d5db0a4bca445a6651d09708041ea/library/core/src/panicking.rs:67:14
   2: core::panicking::panic
             at /rustc/07688726805d5db0a4bca445a6651d09708041ea/library/core/src/panicking.rs:117:5
   3: image::codecs::webp::lossless::LosslessDecoder<R>::read_huffman_code_lengths
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/lossless.rs:438:13
   4: image::codecs::webp::lossless::LosslessDecoder<R>::read_huffman_code
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/lossless.rs:422:17
   5: image::codecs::webp::lossless::LosslessDecoder<R>::read_huffman_codes
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/lossless.rs:366:28
   6: image::codecs::webp::lossless::LosslessDecoder<R>::decode_image_stream
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/lossless.rs:223:28
   7: image::codecs::webp::lossless::LosslessDecoder<R>::read_transforms
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/lossless.rs:269:32
   8: image::codecs::webp::lossless::LosslessDecoder<R>::decode_image_stream
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/lossless.rs:207:13
   9: image::codecs::webp::lossless::LosslessDecoder<R>::decode_frame
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/lossless.rs:160:24
  10: image::codecs::webp::extended::ExtendedImage::read_extended_chunks
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/extended.rs:234:37
  11: image::codecs::webp::decoder::WebPDecoder<R>::read_frame
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/decoder.rs:179:29
  12: image::codecs::webp::decoder::WebPDecoder<R>::read_data
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/decoder.rs:193:21
  13: image::codecs::webp::decoder::WebPDecoder<R>::new
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/codecs/webp/decoder.rs:136:9
  14: image::io::free_functions::load_decoder
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/io/free_functions.rs:63:59
  15: image::io::free_functions::load_inner
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/io/free_functions.rs:113:5
  16: image::io::reader::Reader<R>::decode
             at /DATA2/dev/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/image-0.24.8/src/io/reader.rs:228:9
  17: image_poc::harness
             at ./src/main.rs:11:16
  18: image_poc::main
             at ./src/main.rs:17:2
  19: core::ops::function::FnOnce::call_once
             at /rustc/07688726805d5db0a4bca445a6651d09708041ea/library/core/src/ops/function.rs:250:5

## Reproduction steps

```
use std::io::Cursor;
use image::io::Reader;

fn harness() {

    let p = "image-crash.xx";
    if let Ok(v) = std::fs::read(p) {
        let data = &v;
        if let Ok(mut reader) = Reader::new(Cursor::new(data))
            .with_guessed_format() {
            let image = reader.decode();
        }
    }
}

fn main() {
    harness();
}
```

The input `image-crash.xx` is attached. 

[image-crash.xx.zip](https://github.com/image-rs/image/files/14173754/image-crash.xx.zip)

**Comment by MinghuaWang** [comment]
Test environment:
OS: Ubuntu 20.04, 64 bit
Version: image = "0.24.8"
Target triple: x86_64-unknown-linux-gnu
Rustc version: rustc 1.73.0-nightly (076887268 2023-08-17)

**Comment by fintelia** [comment]
I expect this will be fixed when we switch to `image-webp` for our WebP decoding. If you have a chance, please see whether the same issue happens with that crate

**Comment by Shnatsel** [comment]
Is there a branch of `image` integrating `image-webp` that I can try?

The integration would make it much easier to test against the real-world WebP corpus: https://github.com/image-rs/image-webp/issues/20

**Comment by fintelia** [comment]
@Shnatsel now that #2148 has been merged, the `next-version-0.25` branch uses `image-webp`.

**Comment by Shnatsel** [comment]
This does not happen on the `next-version-0.25` branch - the error is reported without a panic, both in debug and release modes. The fuzzing paid off!
